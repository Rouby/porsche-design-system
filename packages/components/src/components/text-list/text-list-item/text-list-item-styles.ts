import { getCss } from '../../../utils';
import { addImportantToEachRule, hostHiddenStyles } from '../../../styles';
import { spacingStaticMedium } from '@porsche-design-system/utilities-v2';
import { cssVariablePseudoSuffix } from '../text-list/text-list-styles';

export const getComponentCss = (): string => {
  return getCss({
    '@global': addImportantToEachRule({
      ':host': {
        display: 'list-item',
        position: 'relative', // needed by before pseudo-element, used for ordered list
        font: 'inherit', // ensures style can't be overwritten from outside
        color: 'inherit', // ensures style can't be overwritten from outside
        listStyleType: 'inherit', // ensures style can't be overwritten from outside
        paddingLeft: spacingStaticMedium, // space between ::marker/::before and list item
        ...hostHiddenStyles,
      },
      '::slotted(*)': {
        [cssVariablePseudoSuffix]: '""', // don't show suffix "." for nested ordered list
      },
    }),
  });
};
