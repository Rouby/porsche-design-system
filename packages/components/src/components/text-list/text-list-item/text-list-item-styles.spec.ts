import { getComponentCss } from './text-list-item-styles';

describe('getComponentCss()', () => {
  it.each<Parameters<typeof getComponentCss>>([[]])('should return correct css', (...args) => {
    expect(getComponentCss(...args)).toMatchSnapshot();
  });
});
