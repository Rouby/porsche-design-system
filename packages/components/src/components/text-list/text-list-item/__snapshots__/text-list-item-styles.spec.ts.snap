// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css 1`] = `
":host {
  display: list-item !important;
  position: relative !important;
  font: inherit !important;
  color: inherit !important;
  list-style-type: inherit !important;
  padding-left: 16px !important;
}
:host([hidden]) {
  display: none !important;
}
::slotted(*) {
  --p-internal-text-list-pseudo-suffix: "" !important;
}"
`;
